### PR TITLE
remove legacy cmake generator and build helper

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Remove: Remove legacy ``cmake`` generator and ``CMake`` build helper superseded by new ``conan.tools.cmake``
Docs: Omit

### CODE REVIEW

**Comment by memsharded** [comment]
@czoido, the CI is passing https://ci.conan.io/blue/organizations/jenkins/ConanTestSuitev2/detail/PR-9590/11/pipeline, except 2 tests in OSX, if you could please have a look and fix them (and it can be merged if all OK for you)

I have tried to port some of the tests, those that make sense, ``xfail`` those that probably need to be ported, and removed those that were too specific of the legacy ``cmake`` integration.

File: conans/client/command.py [hunk 1]
```diff
@@ -888,7 +888,7 @@ def build(self, *args):
                                      update=args.update, generators=args.generator,
                                      no_imports=args.no_imports,
                                      lockfile=args.lockfile,
-                                     lockfile_out=args.lockfile_out)
+                                     lockfile_out=args.lockfile_out, conf=args.conf_host)
```
Comment (lasote) [inline conans/client/command.py:869]: How is this related?

**Review by lasote — APPROVED** [review]

File: conans/client/command.py (hunk 1 above)
Comment (czoido) [reply conans/client/command.py:869]: This was to fix `test_apple_framework_xcode` test that passes the generator through the command line with `build . -c tools.cmake.cmaketoolchain:generator=Xcode` but the argument was not taken in the build command.

**Review by czoido — APPROVED** [review]
